hbase-client/src/main/java/org/apache/hadoop/hbase/client/RegionCoprocessorRpcChannelImpl.java

```diff
@@ -78,8 +83,7 @@ private CompletableFuture<Message> rpcCall(MethodDescriptor method, Message requ
     final Context context = Context.current();
     CompletableFuture<Message> future = new CompletableFuture<>();
     if (region != null && !Bytes.equals(loc.getRegion().getRegionName(), region.getRegionName())) {
-      future.completeExceptionally(new DoNotRetryIOException("Region name is changed, expected "
-        + region.getRegionNameAsString() + ", actual " + loc.getRegion().getRegionNameAsString()));
+      future.completeExceptionally(new RegionNameChangedException(loc.getRegion()));
```

Comment:
This works, but I wonder if there is a better place to detect this scenario from the client